packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,398 @@
+    /// - was executed, or
+    /// - the voter is not listed.
+    /// @param _proposalId The proposal Id.
+    /// @param _account The address of the user to check.
+    /// @return bool Returns true if the account is allowed to vote.
+    ///@dev The function assumes the queried proposal exists.
+    function canApprove(uint256 _proposalId, address _account) external view returns (bool) {
+        return _canApprove(_proposalId, _account);
+    }
+
+    /// @notice Checks if a proposal can be executed.
+    /// @param _proposalId The ID of the proposal to be checked.
+    /// @return True if the proposal can be executed, false otherwise.
+    function canExecute(uint256 _proposalId) external view returns (bool) {
+        return _canExecute(_proposalId);
+    }
+
+    /// @notice Returns all information for a proposal vote by its ID.
+    /// @param _proposalId The ID of the proposal.
+    /// @return open Wheter the proposal is open or not.
+    /// @return executed Wheter the proposal is executed or not.
+    /// @return parameters The parameters of the proposal vote.
+    /// @return tally The current tally of the proposal vote.
+    /// @return actions The actions to be executed in the associated DAO after the proposal has passed.
+    function getProposal(uint256 _proposalId)
+        public
+        view
+        returns (
+            bool open,
+            bool executed,
+            ProposalParameters memory parameters,
+            Tally memory tally,
+            IDAO.Action[] memory actions
+        )
+    {
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        open = proposal_.open;
+        executed = proposal_.executed;
+        parameters = proposal_.parameters;
+        tally = proposal_.tally;
+        actions = proposal_.actions;
+    }
+
+    /// @notice Internal function to update the minimal approval parameter.
+    /// @param _minApprovals The new minimal approval value.
+    function _updateMinApprovals(uint256 _minApprovals) internal virtual {
+        uint256 addresslistLength_ = addresslistLength(0);
+
+        if (_minApprovals > addresslistLength_) {
+            revert MinApprovalsOutOfBounds({limit: addresslistLength_, actual: _minApprovals});
+        }
+
+        if (_minApprovals < 1) {
+            revert MinApprovalsOutOfBounds({limit: 1, actual: _minApprovals});
+        }
+
+        minApprovals_ = _minApprovals;
+    }
+
+    /// @notice Internal function to approve and, optionally, execute the proposal.
+    /// @param _proposalId The ID of the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the approval cast. The call does not revert if execution is not possible.
+    function _approve(
+        uint256 _proposalId,
+        address _approver,
+        bool _tryExecution
+    ) internal {
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        proposal_.tally.approvals += 1;
+        proposal_.approvers[_approver] = true;
```

Comment:
What if that wallet had already approved? If it is done in an external function, I suggest having all the logic in one place, to avoid missing things that are spread out

Reply:
In the `external` `approve()` method, we do the `_canApprove` check. 
We did it the same way in `AddresslistVoting` and `TokenVoting`, where we have `_vote()`, `vote()`, and `canVote`.
What do you suggest @novaknole?

Reply:
@novaknole is fine with having one method in this case and removing `_approve`.

Removed in 06be91019a55b7b35831c520bc0925b5d8c8f97f